internal/yggdrasil/yggdrasil.go

```diff
@@ -310,33 +317,66 @@ func (h *Handler) updateDeploymentStatuses(oldDeployments []v1alpha1.Deployment,
 	return deployments
 }
 
-func (h *Handler) toWorkloadList(logger logr.Logger, deployments []v1alpha1.EdgeDeployment) models.WorkloadList {
+func (h *Handler) toWorkloadList(ctx context.Context, logger logr.Logger, deployments []v1alpha1.EdgeDeployment, device *v1alpha1.EdgeDevice) (models.WorkloadList, error) {
 	list := models.WorkloadList{}
 	for _, deployment := range deployments {
 		if deployment.DeletionTimestamp != nil {
 			continue
 		}
-		podSpec, err := yaml.Marshal(deployment.Spec.Pod.Spec)
+		spec := deployment.Spec
+		podSpec, err := yaml.Marshal(spec.Pod.Spec)
 		if err != nil {
 			logger.Error(err, "cannot marshal pod specification", "deployment name", deployment.Name)
 			continue
 		}
 		var data *models.DataConfiguration
-		if deployment.Spec.Data != nil && len(deployment.Spec.Data.Paths) > 0 {
+		if spec.Data != nil && len(spec.Data.Paths) > 0 {
 			var paths []*models.DataPath
-			for _, path := range deployment.Spec.Data.Paths {
+			for _, path := range spec.Data.Paths {
 				paths = append(paths, &models.DataPath{Source: path.Source, Target: path.Target})
 			}
 			data = &models.DataConfiguration{Paths: paths}
 		}
+
 		workload := models.Workload{
 			Name:          deployment.Name,
 			Specification: string(podSpec),
 			Data:          data,
 		}
+		authFile, err := h.getAuthFile(ctx, spec.ImageRegistries, deployment.Namespace)
+		if err != nil {
+			h.recorder.Eventf(device, corev1.EventTypeWarning, "Misconfiguration",
```

Comment:
this one is tricky :-) the event will be reported for the device, but in fact, the misconfiguration is on the edgeployment.
this might end with many events for the devices that will try to run the misconfigured deployment.
in terms of context and amount of potentials event, perhaps edgedeployment will be a better fit ?

Reply:
I thought about it a bit and decided to go with the edge device, because that was also the result of more general discussion around events before Ondra's changes. User adding a deployment right now should monitor edgedevice for updates about the deployment on device status and they would also see the events in there; there is no reason to monitor the edgedeployment and adding events there might be confusing at this point.
If we decide to refactor edgedeployment - edgedevice relationship, then maybe at that point we should refactor the events as well.